04/2_print_string_buffer/buffered_print.c

```diff
@@ -0,0 +1,31 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "buffered_print.h"
+
+#define BUFFER_LENGTH (16)
+
+static size_t s_buffer_index = 0;
+static char s_buffer[BUFFER_LENGTH];
+
+void buffered_print(const char* src)
+{
+    while (*src != '\0') {

```

Comment:
i don't think you will get it. so this is 90% complete code that you should just follow. do this and fix any bugs with it.




size_t num_left = strlen(src);

while (num_left > 0)
{
  const size_t num_chars_to_buffer = min(BUFFER_LENGTH - 1 - s_buffer_index, num_left);
  const char* p = src;

  const int buffer_empty = s_buffer_index == 0

  s_buffer_index += num_chars_to_buffer;
  num_left -= num_chars_to_buffer
  if (buffer_empty) {
    strncpy(s_buffer, p, num_chars_to_buffer);
    s_buffer[s_buffer_index] = '\0';  
  } else {
     strncat(s_buffer, p, num_chars_to_buffer);
  }

  if (s_buffer_index == BUFFER_LENGTH-1)  {
     printf()
     s_buffer_index = 0;
  }
}

Reply:
size_t num_left;

num_left = strlen(src);